lib/client.js

```diff
@@ -210,6 +210,136 @@ class Client {
     const relativePathWithQuery = `${params.resourceType}?${new URLSearchParams(params.searchParams)}`;
     return this.httpClient.get(relativePathWithQuery);
   }
+
+  /**
+   * Create a resource.
+   *
+   * @example
+   * const newPatient = {
+   *   resourceType: 'Patient',
+   *   active: true,
+   *   name: [{ use: 'official', family: ['Coleman'], given: ['Lisa', 'P.'] }],
+   *   gender: 'female',
+   *   birthDate: '1948-04-14',
+   * }
+   *
+   * // Using promises
+   * fhirClient.create({
+   *   resource: newPatient,
+   * }).then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.create({
+   *   resource: newPatient,
+   * })
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {String} params.resource - The new resource data to create.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  create(params) {
```

Comment:
I guess regardless this specific implementation, given the inputs to these functions are dissimilar and there wouldn't be much reuse, I'm not seeing what `@typedef` gets us over the conventions around `@param` with sub-params?

Reply:
>How? Each would need a different typedef (maybe trivial re-use, get, delete)

* I think there is a `@typedef` for what is currently called `resource` -- even if we *don't* currently specify its properties.   Developers reading the docs will recognize that it's the "same thing", across methods... regardless of the schema for the particular EHR they are using.  ("intentional code", and all that?)

* I think you have just argued that the existing signature for the `create` method is incorrect:  why is there an intervening `params` Object that encapsulates it?

* The other methods that have a `params` should also have a typedef.  (which `delete` does not use.)

* The `delete` method signature is also probably incorrect:  it should probably be: `delete(resourceType, id)`  
...why does it need a full-blown Object (resource?) representation?

Reply:
> The other methods that have a params should also have a typedef. (which delete does not use.)

Or, we have just discovered why all the method signatures should include explicit `resourceType` and `id` arguments?

Reply:
(except for `create`)

Reply:
> I think there is a @typedef for what is currently called resource -- even if we don't currently specify its properties. Developers reading the docs will recognize that it's the "same thing", across methods... regardless of the schema for the particular EHR they are using. ("intentional code", and all that?)

There are nearly 100 archetypes for FHIR Resources (https://www.hl7.org/fhir/resourcelist.html), that will be handled in another library (I spoke about a future fhir-models library).

> 
> I think you have just argued that the existing signature for the create method is incorrect: why is there an intervening params Object that encapsulates it?

I could have fat-fingered, the point is `@typedef` or `@params` pattern.

> 
> The other methods that have a params should also have a typedef. (which delete does not use.)
> 
> The delete method signature is also probably incorrect: it should probably be: delete(resourceType, id

> ...why does it need a full-blown Object (resource?) representation?


My point is in the end you will have to make a `@typedef` for each FHIR action verb, since all the inputs are different. Because there is little/no reuse, I don't see the benefit over using the `@params` JSDoc approach. But I could be missing something...

Reply:
You wouldn't need to specify every combination:   a `@typedef` Object can refer to other properties that reference another `@typedef` 

Although the `@typedef` might make sense for the existing implementation in this PR as I've previously described -- I now think a couple method signatures are incorrect -- _regardless_ of whether we use typedefs or not.    See:  comments on `create` and `delete` signatures.

Reply:
One of the 1st hits for the google search "javascript method signature arguments object "best practices"

https://codeutopia.net/blog/2016/11/24/best-practices-for-javascript-function-parameters/

Although ^that's obviously not necessarily definitive, it seems to argue that required parameters should be explicit, and not crammed into an Object.

...I'd add: the existing approach forces consumers of our modules to A.) build some Object, and B.) correctly name its properties in order to use these methods -- while: C.) a couple of property names happen to be (usually, but not always?) the same as those in the "resource" whose type we do not want to explicitly nail-down/specify.